File: backend/app/schemas/analytics.py
"""
Analytics schemas for dashboard APIs.
"""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class SummaryMetrics(BaseModel):
    total_policies: int = 0
    active_policies: int = 0
    total_evaluations: int = 0
    avg_confidence: float = 0.0


class DecisionDistribution(BaseModel):
    approve: int = 0
    reject: int = 0
    manual_review: int = 0


class TrendPoint(BaseModel):
    date: str
    count: int


class TopPolicy(BaseModel):
    policy_id: str
    name: str
    match_count: int
    decision: str


class AnalyticsResponse(BaseModel):
    summary: SummaryMetrics
    distribution: DecisionDistribution
    trends: dict[str, list[TrendPoint]]
    top_policies: list[TopPolicy]

File: frontend/src/types/analytics.ts
export interface AnalyticsSummary {
  total_policies: number
  active_policies: number
  total_evaluations: number
  avg_confidence: number
  policy_status_breakdown?: Record<string, number>
}

export interface DecisionDistribution {
  approve: number
  reject: number
  manual_review: number
}

export interface TrendPoint {
  date: string
  count: number
}

export interface AnalyticsTrends {
  evaluations?: TrendPoint[]
  APPROVE?: TrendPoint[]
  REJECT?: TrendPoint[]
  MANUAL_REVIEW?: TrendPoint[]
  [key: string]: TrendPoint[] | undefined
}

export interface TopPolicy {
  policy_id: string
  name: string
  match_count: number
  decision: string
}

export interface AnalyticsDashboard {
  summary: AnalyticsSummary
  distribution: DecisionDistribution
  trends: AnalyticsTrends
  top_policies: TopPolicy[]
}
